pkg/granted/credential_process.go

```diff
@@ -24,28 +25,45 @@ type awsCredsStdOut struct {
 var CredentialProcess = cli.Command{
 	Name:  "credential-process",
 	Usage: "Exports AWS session credentials for use with AWS CLI credential_process",
-	Flags: []cli.Flag{&cli.StringFlag{Name: "profile", Required: true}, &cli.StringFlag{Name: "url"}},
+	Flags: []cli.Flag{
+		&cli.StringFlag{Name: "profile", Required: true},
+		&cli.StringFlag{Name: "url"},
+		&cli.DurationFlag{Name: "window", Value: 15 * time.Minute},
+	},
 	Action: func(c *cli.Context) error {
 
 		profileName := c.String("profile")
-		profiles, err := cfaws.LoadProfilesFromDefaultFiles()
-		if err != nil {
-			return err
-		}
 
-		profile, err := profiles.LoadInitialisedProfile(c.Context, profileName)
+		secureSessionCredentialStorage := securestorage.NewSecureSessionCredentialStorage()
+		creds, ok, err := secureSessionCredentialStorage.GetCredentials(profileName)
 		if err != nil {
 			return err
 		}
 
-		duration := time.Hour
-		if profile.AWSConfig.RoleDurationSeconds != nil {
-			duration = *profile.AWSConfig.RoleDurationSeconds
-		}
+		if !ok || (creds.CanExpire && time.Now().After(creds.Expires.Add(-c.Duration("window")))) {
```

Comment:
We can add a bit of debug logging here to help users diagnose caching issues by running
```
GRANTED_LOG=debug granted credential-process --profile demo
```

We use our own CLI logging package [clio](https://github.com/common-fate/clio) for this:

```suggestion
		var needsRefresh bool

		now := time.Now()
		refreshTime := creds.Expires.Add(-c.Duration("window"))

		if !ok {
			clio.Debugw("refreshing credentials", "reason", "not found")
			needsRefresh = true
		} else if creds.CanExpire && now.After(refreshTime) {
			clio.Debugw("refreshing credentials", "reason", "after refresh time", "now", now.String(), "refresh_time", refreshTime.String())
			needsRefresh = true
		}

		if needsRefresh {
```